tests/functionaltests/Tests/ToastNotificationTests.mm

```diff
@@ -0,0 +1,194 @@
+//******************************************************************************
+//
+// Copyright (c) Microsoft. All rights reserved.
+//
+// This code is licensed under the MIT License (MIT).
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//******************************************************************************
+
+#include <COMIncludes.h>
+#include "MockClass.h"
+#include <windows.applicationModel.activation.h>
+#include <COMIncludes_End.h>
+
+#include <UWP/WindowsApplicationModelActivation.h>
+#include <TestFramework.h>
+#include <Foundation/Foundation.h>
+#include "UIKit/UIApplication.h"
+#include "StringHelpers.h"
+
+using namespace ABI::Windows::ApplicationModel::Activation;
+using namespace Microsoft::WRL;
+
+// Method to call in tests to activate app
+extern "C" void UIApplicationActivationTest(IInspectable* args, void* delegateClassName);
+
+MOCK_CLASS(MockToastNotificationActivatedEventArgs,
+           public RuntimeClass<RuntimeClassFlags<WinRtClassicComMix>, IToastNotificationActivatedEventArgs, IActivatedEventArgs> {
+
+               // Claim to be the implementation for the real system RuntimeClass for ToastNotificationActivatedEventArgs.
+               InspectableClass(RuntimeClass_Windows_ApplicationModel_Activation_ToastNotificationActivatedEventArgs, BaseTrust);
+
+           public:
+               MOCK_STDCALL_METHOD_1(get_Argument);
+               MOCK_STDCALL_METHOD_1(get_UserInput);
+               MOCK_STDCALL_METHOD_1(get_Kind);
+               MOCK_STDCALL_METHOD_1(get_PreviousExecutionState);
+               MOCK_STDCALL_METHOD_1(get_SplashScreen);
+           });
+
+@interface ToastNotificationForegroundActivationTestDelegate : NSObject <UIApplicationDelegate>
+@property (nonatomic, readonly) NSMutableDictionary* methodsCalled;
+@end
+
+@implementation ToastNotificationForegroundActivationTestDelegate
+- (id)init {
+    self = [super init];
+    if (self) {
+        _methodsCalled = [NSMutableDictionary new];
+    }
+    return self;
+}
+
+- (BOOL)application:(UIApplication*)application willFinishLaunchingWithOptions:(NSDictionary*)launchOptions {
+    ASSERT_TRUE(launchOptions[UIApplicationLaunchOptionsToastNotificationKey]);
+    WAAToastNotificationActivatedEventArgs* args = launchOptions[UIApplicationLaunchOptionsToastNotificationKey];
+    ASSERT_STREQ("TOAST_NOTIFICATION_TEST", [args.argument UTF8String]);
```

Comment:
>TREQ("TOAST_NOTIFICATION_TEST", [args.argument UTF8String]); [](start = 12, length = 60)

universally, `ASSERT_STREQ("a", [b UTF8String])` can be better represented by `ASSERT_OBJCEQ(@"a", b)`.